similaritymeasures/similaritymeasures.py

```diff
@@ -879,3 +879,125 @@ def dtw_path(d):
     path = np.array(path)
     # reverse the order of path, such that it starts with [0, 0]
     return path[::-1]
+
+def mae(exp_data, num_data, metric='euclidean', **kwargs):
+    """
+    Compute the Mean Absolute distance.
+
+    This computes the mean of absolute values of the distances 
+    between 2 corresponding points on the curve. This can use all 
+    distance metrics that are available in scipy.spatial.distance.cdist.
+
+    Parameters
+    ----------
+    exp_data : array_like
+        Curve from your experimental data. exp_data is of (M, N) shape, where
+        M is the number of data points, and N is the number of dimmensions
+    num_data : array_like
+        Curve from your numerical data. num_data is of (P, N) shape, where P
+        is the number of data points, and N is the number of dimmensions
+    metric : str or callable, optional
+        The distance metric to use. Default='euclidean'. Refer to the
+        documentation for scipy.spatial.distance.cdist. Some examples:
+        'braycurtis', 'canberra', 'chebyshev', 'cityblock', 'correlation',
+        'cosine', 'dice', 'euclidean', 'hamming', 'jaccard', 'kulsinski',
+        'mahalanobis', 'matching', 'minkowski', 'rogerstanimoto', 'russellrao',
+        'seuclidean', 'sokalmichener', 'sokalsneath', 'sqeuclidean',
+        'wminkowski', 'yule'.
+    **kwargs : dict, optional
+        Extra arguments to `metric`: refer to each metric documentation in
+        scipy.spatial.distance.
+
+        Some examples:
+
+        p : scalar
+            The p-norm to apply for Minkowski, weighted and unweighted.
+            Default: 2.
+
+        w : ndarray
+            The weight vector for metrics that support weights (e.g.,
+            Minkowski).
+
+        V : ndarray
+            The variance vector for standardized Euclidean.
+            Default: var(vstack([XA, XB]), axis=0, ddof=1)
+
+        VI : ndarray
+            The inverse of the covariance matrix for Mahalanobis.
+            Default: inv(cov(vstack([XA, XB].T))).T
+
+        out : ndarray
+            The output array
+            If not None, the distance matrix Y is stored in this array.
+
+    Retruns
+    -------
+    r : float
+        MAE.
+    """
+    c = distance.cdist(exp_data, num_data, metric=metric, **kwargs)
+    mae = 0
+    for i in range(0, len(c)):
+        mae += abs(c[i, i])
+    mae /= len(c)
```

Comment:
Since these are all numpy arrays, it's better to call numpy methods like `c.sum()` and `c.mean()`. You generally don't want to use a for loop to go through elements of a numpy array.